src/app/shared/auth.interceptor.ts

```diff
@@ -35,20 +44,45 @@ export class AuthInterceptor implements HttpInterceptor {
       .pipe(
         tap(
           // Succeeds when there is a response; ignore other events
-          event => ok = event instanceof HttpResponse ? 'succeeded' : '',
+          event => event instanceof HttpResponse ? this.refreshRPT(event) : '',
           // Operation failed; error is an HttpErrorResponse
-          error => {
-            if (error instanceof HttpErrorResponse) {
-              const res: HttpErrorResponse = error;
-              if (res.status === 403 || isAuthenticationError(res)) {
-                this.broadcaster.broadcast('authenticationError', res);
-              } else if (res.status === 500) {
-                this.broadcaster.broadcast('communicationError', res);
-              }
-            }
-            return throwError(error);
-          }
+          error => error instanceof HttpErrorResponse ? this.catchError(error) : throwError(error)
         )
       );
   }
+
+  private refreshRPT(res: HttpResponse<any>) {
+    if (res.headers.has('Authorization')) {
+      const token = localStorage.getItem('auth_token');
+      const newToken = res.headers.get('Authorization').replace('Bearer ', '');
+      if (token !== newToken) {
+        localStorage.setItem('auth_token', newToken);
+      }
+    }
+  }
+
+  private catchError(res: HttpErrorResponse) {
+    if (res.status === 403 || this.isAuthenticationError(res)) {
+      this.broadcaster.broadcast('authenticationError', res);
+    } else if (res.status === 500) {
+      this.broadcaster.broadcast('communicationError', res);
+    }
+  }
+
+  private isAuthenticationError(res: HttpErrorResponse): boolean {
```

Comment:
If we don't need `isAuthenticationError` then why was it moved in here?

Reply:
We do use `isAuthenticationError` but only inside the interceptor and that is why i moved it here. Earlier it was being used by other services as well.

Reply:
I'm not understanding why make the change. If it is external it keeps the code clean, it can be tested and it can be shared. It is more functional to keep it external. 

What is the benefit of moving it inside?

Reply:
It will not be used anywhere else since interceptor does that already for every response. So, it made more sense to keep it tightly coupled as a bussiness logic to auth interceptor rather than keeping it external.